### ISSUE TITLE
prefix and separator optional

### ISSUE THREAD

**Issue by mbklaene** [flyway/flyway#2797]
###### Which version and edition of Flyway are you using?

6.1.4

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)

command-line

###### Which database are you using (type & version)?

postgres 10

###### Which operating system are you using?

ubuntu

###### What did you do?

I tried to set the repeatable prefix to a blank value as well as the separator. I received an error that the separator cannot be blank.

What I would like to do is provide a way for users to use the repeatable approach to migrations but not supply a prefix or separator. 
repeatableSqlMigrationPrefix=''
sqlMigrationSeparator=''

In other words, '/sql/mytables.sql' would always be seen as a repeatable migration.   In this scenario, versioned migrations would not be used.

Can you advise if this can be achieved through some set of configurations?

**Comment by MikielAgutu** [comment]
This is by design. Flyway enforces the filename convention in order to determine the nature of the migrations. This isn't really something we want to change.

Can you explain more about why you need to do this?

**Comment by mbklaene** [comment]
Thanks. The reason is some users would simply just prefer to put files in an 'sql' directory and know that they will be executed. Essentially, it's the repeatable pattern, without the prefix/separator.

I think it would be a good option if someone wanted to 'turn off' the versioned migrations (deliberately, with a config setting). And if they did that, there would really be no point in having the repeatable prefix. It could just be assumed.

**Comment by MikielAgutu** [comment]
I take your point. For us it's important for Flyway to enforce this naming convention. In the vast majority of cases users require repeatable and versioned migrations.

I agree it's a bit annoying to write the prefix if you only need repeatables. Nonetheless I think it's a reasonable compromise.
